src/Connection.php

```diff
@@ -1372,6 +1355,28 @@ public function commit()
         return $result;
     }
 
+    /**
+     * @return bool
+     *
+     * @throws DriverException
+     */
+    private function doCommit(DriverConnection $connection)
```

Comment:
```suggestion
    /**
     * @throws DriverException
     */
    private function doCommit(DriverConnection $connection): bool
```

Reply:
I did not want to do it because it would require modifying a large amounts of mocks in tests. Current driver connection mocks don't return for `$connection->commit()`.

That's why I tried to do https://github.com/doctrine/dbal/pull/5685 With that PR phpunit mocks were able to return `bool` from driver `connection->commit()` so it would have satisfied this type hint.

Reply:
OK 👍

Reply:
> I did not want to do it because it would require modifying a large amounts of mocks in tests.

Which would also mean a BC break.